sql-plugin/src/main/scala/com/nvidia/spark/rapids/basicPhysicalOperators.scala

```diff
@@ -107,8 +107,13 @@ object GpuProjectExec {
       // This can help avoid contiguous splits in some cases when the input data is also contiguous
       GpuColumnVector.incRefCounts(cb)
     } else {
-      val newColumns = boundExprs.safeMap(_.columnarEval(cb)).toArray[ColumnVector]
-      new ColumnarBatch(newColumns, cb.numRows())
+      try {
+        GpuExpressionsUtils.cachedNullVectors.get.clear()
```

Comment:
There are more ways than just this method to run a expression. I don't trust this to fix it every time. It is probably god enough in practice, but I don't like the precedence it is setting. At a minimum I want to see some comments here explaining what is happening and why. Preferably with a follow on issue to fix this once we have the ability to delete a buffer from the spill framework instead of spilling it.